### DISCUSSION TITLE
subpaths

### DISCUSSION THREAD

**Discussion by AmirL** [trpc/trpc#3582] [Q&A] [answered]
Hello,

I'm trying to figure out how to import sub-routes, but faced a problem. I've made a [simple sandbox with this problem](https://codesandbox.io/p/github/AmirL/trpc-server-test/draft/nostalgic-fermat?create=true&file=%2Findex.ts&selection=%5B%7B%22endColumn%22%3A9%2C%22endLineNumber%22%3A17%2C%22startColumn%22%3A9%2C%22startLineNumber%22%3A17%7D%5D&workspace=%257B%2522activeFileId%2522%253A%2522clcp1juy500017qi7b85496zj%2522%252C%2522openFiles%2522%253A%255B%2522%252Findex.ts%2522%252C%2522%252Froutes%252Ftest.ts%2522%255D%252C%2522sidebarPanel%2522%253A%2522GIT%2522%252C%2522gitSidebarPanel%2522%253A%2522COMMIT%2522%252C%2522spaces%2522%253A%257B%2522clcp1jwq7000e3d6gmu7fw6k1%2522%253A%257B%2522key%2522%253A%2522clcp1jwq7000e3d6gmu7fw6k1%2522%252C%2522name%2522%253A%2522Default%2522%252C%2522devtools%2522%253A%255B%257B%2522type%2522%253A%2522PREVIEW%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A2022%252C%2522key%2522%253A%2522clcp1k464007y3d6gcx0tf4ti%2522%252C%2522isMinimized%2522%253Afalse%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522key%2522%253A%2522clcp1k2xi004l3d6gk6t1ybpg%2522%252C%2522isMinimized%2522%253Afalse%257D%255D%257D%257D%252C%2522currentSpace%2522%253A%2522clcp1jwq7000e3d6gmu7fw6k1%2522%252C%2522spacesOrder%2522%253A%255B%2522clcp1jwq7000e3d6gmu7fw6k1%2522%255D%257D).

I have 2 questions about it:
1) Why I get error: 
ReferenceError: Cannot access 'router' before initialization
    at Object.router (/project/trpc-server-test/index.ts:1:1)
? I'm looking at https://trpc.io/docs/merging-routers and it seems the same to me.

2) If I write this router inside index.ts ( I commented it there), I can't access it by "server.url/test/hello". But it works with "server.url/test.hello". Which is strange to me. 

I'd like to get some help, cause I don't see what I'm doing wrong.

**Comment by KATT** [comment, marked as answer]
You're importing `import { testRouter } from "./routes/test";` ahead of defining `export const router = t.router;`

This creates a circular dep that JS will go a bit nuts on. Create a file called `trpc.ts` with the initialization of trpc

**Reply by AmirL** [reply]
Thanks! I've create a trpc.ts file and it solved the problem. 

Still, I haven't figured out about the path. Is it supposed to work in this way:
https://mnpt2t-2022.preview.csb.app/test.hello

instead of:
https://mnpt2t-2022.preview.csb.app/test/hello

I read all the docs on trpc.io and no mentions of subpaths there. 

PS
In case someone will read it later:
```
// trpc.ts
import { initTRPC } from "@trpc/server";
export const trpc = initTRPC.create();

export const publicProcedure = trpc.procedure;
export const router = trpc.router;
```

**Reply by juliusmarminge** [reply]
All procedures go to your main api handler, e.g. `http://localhost:3000/api/trpc`

Then the query parameters after that will include procedure paths, batching, inputs etc
